### PR TITLE
Etl and Ingest object buffered tiles reprojection

### PR DESCRIPTION
**Pull request by pomadchin**

### CODE REVIEW

File: spark-etl/src/main/scala/geotrellis/spark/etl/EtlConf.scala [hunk 1]
```diff
@@ -48,6 +48,8 @@ class EtlConf(args: Seq[String]) extends ScallopConf(args){
   val tileSize     = opt[Int]("tileSize",
                       descr = "Tile size in pixels (default: 256)",
                       default = Some(256))
+  val bufferSize   = opt[Int]("bufferSize",
```
Comment (echeipesh) [inline spark-etl/src/main/scala/geotrellis/spark/etl/EtlConf.scala]: Is there a reason for user to be able to control this?

Comment (pomadchin) [inline spark-etl/src/main/scala/geotrellis/spark/etl/EtlConf.scala]: To be honest, not sure, think just preventing of its calculation depending on tiles metadata. Probably for `Etl` we can remove it, but left for the `Ingest` object.

File: raster/src/main/scala/geotrellis/raster/Tile.scala [hunk 2]
```diff
@@ -292,4 +292,10 @@ trait Tile extends CellGrid with IterableTile with MappableTile[Tile] {
     }
     s
   }
+
+  def isNoDataTile = {
+    var empty = true
+    foreach { z => if (isData(z)) empty = false }
```
Comment (echeipesh) [inline raster/src/main/scala/geotrellis/raster/Tile.scala]: There should be a way to break out of this early if we're going to provide this method.

Comment (pomadchin) [inline raster/src/main/scala/geotrellis/raster/Tile.scala]: Have you an idea how to call it? I like the `isEmpty`, but not sure that it is a correct function name and can be confusing.

Comment (echeipesh) [inline raster/src/main/scala/geotrellis/raster/Tile.scala]: +@lossyrob I think the name is appropriate. I remember wanting a function like this but stopped just short of committing it.

**Comment by echeipesh** [comment]
+1

File: raster/src/main/scala/geotrellis/raster/Tile.scala (hunk 2 above)
Comment (lossyrob) [inline raster/src/main/scala/geotrellis/raster/Tile.scala]: The much faster way to do this is to have a return inside of a while loop, or some other way to break immediately upon finding a non-nodata value

**Comment by lossyrob** [comment]
I believe this was still in progress as we were figuring out where the difference in NODATA tiles was coming from with modifying buffered tiles. @pomadchin if there's updates with that, those can be in a separate PR.

File: raster/src/main/scala/geotrellis/raster/Tile.scala (hunk 2 above)
Comment (pomadchin) [inline raster/src/main/scala/geotrellis/raster/Tile.scala]: @lossyrob ok will make a probably small pr, yes throwing an exception can be significant :/

**Comment by pomadchin** [comment]
@lossyrob:+1: mb it is even better to perform it as a separate pr (if there would be a pr at all)

**Pull request merged**